packages/ui-library/src/components/icon-menu/index.tsx

```diff
@@ -63,16 +99,40 @@ const IconMenu = ({ options, color = 'info', variant = 'text', icon, size = 'sma
           horizontal: 'left',
         }}
       >
-        {options.map(({ onClick, icon: itemIcon, label, control, closeOnClick = true }) => (
-          <MenuItem key={label} onClick={() => handleCloseWithAction(closeOnClick, onClick)}>
-            {itemIcon}
-            {label}
-            {control}
-          </MenuItem>
-        ))}
+        {options.map((option, index) => {
+          if (option.type === IconMenuOptionType.option) {
+            const {
+              label,
+              closeOnClick = true,
+              color: itemColor,
+              control,
+              onClick,
+              icon: itemIcon,
+              secondaryLabel,
+            } = option;
+            return (
+              <MenuItem
+                onClick={() => handleCloseWithAction(closeOnClick, onClick)}
+                style={{ color: itemColor }}
+                key={index}
+              >
+                <Box>
```

Comment:
I'm not sure I understand the need for boxes inside this component, they don't provide any styling

Reply:
I have added styles for this box in the components.ts file for all Mui MenuItems. In this case, this box needs to align the Icon with the 1 or 2 labels in the  MenuItem. As a sibling to this box, there is the `control` that need to be positioned at the end to the right

Reply:
Just tried it and we can safely remove this first box

Reply:
Just add the align items to the MuiMenuItem

Reply:
Thank you for giving this a chance! I have moved some styles to the MuiMenuItem global styles as you suggested. Regarding of aligning the items, the `control` icon needs to be positioned at the end of the menuItem. I found a new way to archive this without the <Box> wrapper as it was.